### ISSUE TITLE
ERRO Failed to query DNS from host: lookup api.crc.testing: no such host

### ISSUE THREAD

**Issue by kevprice83** [code-ready/crc#1445]
OS Name:                   Microsoft Windows 10 Home
OS Version:                10.0.19041 N/A Build 19041
OS Manufacturer:           Microsoft Corporation
OS Configuration:          Standalone Workstation
OS Build Type:             Multiprocessor Free
Hypervisor: Hyper-V
  * Did you run `crc setup` before starting it (Yes/No)?
Yes
  * Running CRC on: Laptop

## CRC version
```bash
CodeReady Containers version: 1.14.0+36ad776
OpenShift version: 4.5.4 (embedded in binary)
```

The following is run after the failure so it's expected at this point.

## CRC status
```bash
DEBU CodeReady Containers version: 1.14.0+36ad776
DEBU OpenShift version: 4.5.4 (embedded in binary)
DEBU
ERRO Cannot get root partition usage: Error getting config for native Go SSH: open C:\Users\kpric\.crc\machines\crc\id_rsa: The system cannot find the file specified.
Cannot get root partition usage: Error getting config for native Go SSH: open C:\Users\kpric\.crc\machines\crc\id_rsa: The system cannot find the file specified.
```

## CRC config
```bash
- nameserver                            : 8.8.8.8
```

## Host Operating System
```bash
Host Name:                 KEVIN-LENOVO
OS Name:                   Microsoft Windows 10 Home
OS Version:                10.0.19041 N/A Build 19041
OS Manufacturer:           Microsoft Corporation
OS Configuration:          Standalone Workstation
OS Build Type:             Multiprocessor Free
Registered Owner:          <redacted>
Registered Organization:
Product ID:                00326-10000-00000-AA723
Original Install Date:     07/07/2020, 20:26:04
System Boot Time:          07/08/2020, 18:34:56
System Manufacturer:       LENOVO
System Model:              20EQS64N09
System Type:               x64-based PC
Processor(s):              1 Processor(s) Installed.
                           [01]: Intel64 Family 6 Model 94 Stepping 3 GenuineIntel ~2712 Mhz
BIOS Version:              LENOVO N1EET89W (1.62 ), 18/06/2020
Windows Directory:         C:\WINDOWS
System Directory:          C:\WINDOWS\system32
Boot Device:               \Device\HarddiskVolume2
System Locale:             en-gb;English (United Kingdom)
Input Locale:              es;Spanish (Traditional Sort)
Time Zone:                 (UTC+01:00) Brussels, Copenhagen, Madrid, Paris
Total Physical Memory:     32,663 MB
Available Physical Memory: 7,832 MB
Virtual Memory: Max Size:  37,527 MB
Virtual Memory: Available: 8,910 MB
Virtual Memory: In Use:    28,617 MB
Page File Location(s):     C:\pagefile.sys
Domain:                    WORKGROUP
Logon Server:              \\KEVIN-LENOVO
                           [01]: KB4565627
                           [02]: KB4560366
                           [03]: KB4561600
                           [04]: KB4566785
                           [05]: KB4565503
Network Card(s):           10 NIC(s) Installed.
                           [01]: TAP-NordVPN Windows Adapter V9
                                 Connection Name: Ethernet 2
                                 Status:          Media disconnected
                           [02]: Intel(R) Dual Band Wireless-AC 8260
                                 Connection Name: WiFi
                                 DHCP Enabled:    Yes
                                 DHCP Server:     192.168.1.1
                                 IP address(es)
                                 [01]: 192.168.1.108
                                 [02]: fe80::74d8:5382:4516:349
                                 [03]: 2a01:c50e:2181:a300:bd31:29ec:9722:6083
                                 [04]: 2a01:c50e:2181:a300:74d8:5382:4516:349
                           [03]: Intel(R) Ethernet Connection (2) I219-LM
                                 Connection Name: Ethernet
                                 Status:          Media disconnected
                           [04]: Nlwt Tun
                                 Connection Name: NordLynx
                           [05]: Bluetooth Device (Personal Area Network)
                                 Connection Name: Bluetooth Network Connection
                                 Status:          Media disconnected
                           [06]: TAP-Windows Adapter V9
                                 Connection Name: Local Area Connection 2
                                 Status:          Media disconnected
                           [07]: TAP-Windows Adapter V9
                                 Connection Name: Local Area Connection 3
                                 Status:          Media disconnected
                           [08]: Hyper-V Virtual Ethernet Adapter
                                 Connection Name: vEthernet (Default Switch)
                                 DHCP Enabled:    No
                                 IP address(es)
                                 [01]: 172.21.112.1
                                 [02]: fe80::b89e:1878:5f17:2084
                           [09]: Hyper-V Virtual Ethernet Adapter
                                 Connection Name: vEthernet (WSL)
                                 DHCP Enabled:    No
                                 IP address(es)
                                 [01]: 172.26.112.1
                                 [02]: fe80::e480:5b28:5c66:b6a2
                           [10]: Microsoft KM-TEST Loopback Adapter
                                 Connection Name: Ethernet 4
                                 DHCP Enabled:    Yes
                                 DHCP Server:     255.255.255.255
                                 IP address(es)
                                 [01]: 169.254.160.5
                                 [02]: fe80::a4b6:7307:6e3:a005
Hyper-V Requirements:      A hypervisor has been detected. Features required for Hyper-V will not be displayed.
```

### Steps to reproduce

  1. crc setup
  2. crc start
  

### Expected
both internal and external DNS queries are successful

### Actual
internal DNS resolution fails

### Logs

[logs](https://gist.github.com/kevprice83/714d95b0fdbaf97e8e5378a2cb4932ea)

### Comments

I managed to workaround this issue by capturing the IP of the VM before the DNS checks in the script happen and I added them in the etc/hosts file. This obviously points to an issue with how DNS resolution is happening with the default switch but I'm using a clean environment where no networking configuration has ever been done previously. There is some configuration steps missing from the deployment guide I feel.

I already tried creating a new switch named `crc` as suggested in other posts but this didn't resolve the issue. Bizarrely after removing that switch last night and reverting to the default switch the startup succeeded where it normally failed and then failed on public DNS resolution. I was able to fix that by setting a nameserver in the crc config. However, I left the cluster running over night and it went into a _degraded_ state and I am now back to the same situation.

Is there something I might be missing with the networking configuration that needs to be done?

*Other important notes*: I use WSL2 + Docker Desktop. I am not aware of the feature LCOW and whether that is enabled, I cannot find it anywhere in my settings.

**Comment by kevprice83** [comment]
Other things I've checked:

Firewall logs show no dropped packets on DNS and TCP protocols for the IPs associated to the VM.

Switched off all Windows Defender Firewall settings and still the same result.

**Comment by kevprice83** [comment]
It looks like there is still an unresolved issue regarding the expected DNS configuration on Windows as mentioned in #756 . That issue links back to #751 where you can see the same workaround has to be applied as I was doing with the _etc/hosts_ file. So aside from what seems to be missing documentation on DNS setup in Windows I think clearly there is some differences between certain Windows environments given that you are unable to reproduce this @praveenkumar . Is it possible to look at the Switch settings you are using to see if we can find some differences there? I am using the default switch and did not touch anything there, it has DHCP disabled by default and I see that it gets an IP and subnet mask assigned whilst the DNS entries are empty until the CRC VM is started up at which point the first DNS entry is populated with the IP of the VM. This to me seems expected so I'm not sure what is wrong here.

I have tried enabling DHCP on a custom switch called `crc`. The IP was set automatically once I shared my WiFi network with that switch and I populated the DNS entry with the IP of the VM as soon as it started up. It still failed with the following:

```
INFO Check DNS query from host ...
DEBU api.crc.testing resolved to [172.20.224.226]
ERRO Failed to query DNS from host: lookup foo.apps-crc.testing: no such host
Failed to query DNS from host: lookup foo.apps-crc.testing: no such host
```

Notice how the first internal DNS is resolved successfully on api.crc.testing but the next one on foo.apps-crc.testing fails. Why is this and what does CRC use that domain for?

**Comment by gbraad** [comment]
Unfortunately we do not support deploying CRC on Windows 10 Home, until Microsoft officially launches the use of Hyper-V on this edition. When patched or modified we can not guarantee this works identical to the Pro edition of Windows due to other dependencies that might exist.

**Comment by par02** [comment]
@gbraad The same behaviour is seen on Windows 10 Enterprise with Hyper-V using the current version of CRC.
```
OS Name:                   Microsoft Windows 10 Enterprise
OS Version:                10.0.17134 N/A Build 17134

CodeReady Containers version: 1.14.0+36ad776
OpenShift version: 4.5.4 (embedded in binary)
```
Out-of-the-box, the same DNS query failure occurs for api.crc.testing.
To get around this I used the same workaround as kevprice83, namely to edit the Windows 'hosts' file while 'crc start' was running to add the VM's IP address for the necessary mappings (api.crc.testing console-openshift-console.apps-crc.testing oauth-openshift.apps-crc.testing).

There is only a small window to do this, maybe 30 seconds between Hyper-V assigning an IP address and the start script reaching 'Check DNS query from host ...'
It has to be re-edited each time it is started as well, since Hyper-V doesn't always give the same address.

**Comment by gbraad** [comment]
The reporter has been instructed to look into another issue to see if this helps: https://github.com/code-ready/crc/issues/1193

This was closed as being reported for Home. We have discussed this in an online session.
